docker/start.py

```diff
@@ -205,41 +205,53 @@ def main(args, environ):
             config_dir, config_path, environ, ownership
         )
 
-    if mode is not None:
-        error("Unknown execution mode '%s'" % (mode,))
+    if mode == "run":
+        args = args[2:]
 
-    config_dir = environ.get("SYNAPSE_CONFIG_DIR", "/data")
-    config_path = environ.get("SYNAPSE_CONFIG_PATH", config_dir + "/homeserver.yaml")
+        if "-m" not in args:
+            args = ["-m", synapse_worker] + args
 
-    if not os.path.exists(config_path):
-        if "SYNAPSE_SERVER_NAME" in environ:
-            error(
-                """\
-Config file '%s' does not exist.
+        # if there are no config files passed to synapse, try adding the default file
+        if not any(p.startswith("--config-path") for p in args):
```

Comment:
we should probably also check for `-c` here.

Reply:
Yes, missed that.